lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -331,21 +326,30 @@ withNetworkLayerBase tr np conn (versionData, _) tol action = do
     let readCurrentNodeEra = atomically $ readTMVar eraVar
 
     action $ NetworkLayer
-        { currentNodeTip =
+        { chainSync = \follower -> do
+            client <- mkWalletClient
+                (contramap MsgChainSyncCmd tr)
+                (mapChainFollower (toPoint getGenesisBlockHash) (fromTip' gp) id follower)
+                cfg
+
+            let handlers = failOnConnectionLost tr
+            --let points = reverse $ genesisPoint :
+            --        (toPoint getGenesisBlockHash <$> headers)
+
+            -- TODO: Trace
+            -- traceWith tr $ MsgFindIntersection headers
+            -- $ MsgIntersectionFound
+            --
+            -- TODO: Recover on connection lost exceptions!
+            connectClient tr handlers client versionData conn
+
+        , currentNodeTip =
             fromTip getGenesisBlockHash <$> atomically readNodeTip
         , currentNodeEra =
             -- NOTE: Is not guaranteed to be consistent with @currentNodeTip@
             readCurrentNodeEra
         , watchNodeTip = do
```

Comment:
With the reformulation of chainSync, this function will probably be redundant.

Reply:
I don't see how `watchNodeTip` would be related to this PR.

We use it for
1. Caching reward balances in DB
2. Tx resubmission

It would be good to unify the DB and in-memory caching of rewards, but that's a separate concern. (I think this might contribute to values being slower to be updated, and perhaps to some integration test flakiness).

Reply:
I guess you mean we could implement `watchNodeTip` using `chainSync`?

But I think 
1) having a single tip-sync protocol and multiple `STM` observers (allow callbacks to be skipped under load)
2) not having to deserialise block bodies

are good qualities with the current approach, plus I don't see how `ChainFollower` could allow fast-forwarding the intersection to the tip.